Jenkinsfile

```diff
@@ -15,7 +15,7 @@ def mavenEnv(Map params = [:], Closure body) {
                 // Exclude DigitalOcean artifact caching proxy provider, currently unreliable on BOM builds
                 // TODO: remove when https://github.com/jenkins-infra/helpdesk/issues/3481 is fixed
                 infra.withArtifactCachingProxy(env.ARTIFACT_CACHING_PROXY_PROVIDER != 'do') {
-                    withEnv(["MAVEN_ARGS=-Dmaven.repo.local=${WORKSPACE_TMP}/m2repo"]) {
+                    withEnv(["MAVEN_ARGS=-B -ntp ${env.MAVEN_SETTINGS != null ? /-s $MAVEN_SETTINGS/ : ''} -Dmaven.repo.local=${WORKSPACE_TMP}/m2repo"]) {
```

Comment:
Rather than defining this arbitrary variable, `infra.withArtifactCachingProxy` could simply bind `MAVEN_ARGS` directly. That would make the settings file be used automatically:

```groovy
infra.withArtifactCachingProxy {
  sh 'mvn install'
}
```

Reply:
https://github.com/jenkins-infra/pipeline-library/pull/645 is merged so I could try to use that.